### PR TITLE
[REF] web, spreadsheet: use MultiRecordSelector instead of RecordsSelector

### PR DESCRIPTION
**Pull request by pro-odoo**
These commits replace the use of `RecordsSelector` by the `MultiRecordSelector`.
To be able to achieve this, we have to enhance the behavior of `MultiRecordSelector` with the support of placeholder and the support of the keyboard navigation

### CODE REVIEW

File: addons/web/static/tests/core/record_selectors/multi_record_selector_tests.js [hunk 1]
```diff
@@ -163,4 +163,24 @@ QUnit.module("Web Components", (hooks) => {
...
     });
+
+    QUnit.test("Support placeholder", async (assert) => {
+        await makeMultiRecordSelector({
+            resModel: "partner",
+            resIds: [],
+            placeholder: "Select a partner",
+        });
+        const input = target.querySelector(".o_multi_record_selector input");
+        assert.strictEqual(input.placeholder, "Select a partner");
```
Comment (FrancoisGe) [inline addons/web/static/tests/core/record_selectors/multi_record_selector_tests.js:174]: add: Select a record and check if the placeholder is remove ?

File: addons/web/static/src/core/record_selectors/tag_navigation_hook.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    const ref = useRef(refName);
+
+    useEffect(
+        (autocomplete) => {
+            autocomplete?.addEventListener("keydown", onAutoCompleteKeydown);
+            return () => {
+                autocomplete?.removeEventListener("keydown", onAutoCompleteKeydown);
+            };
+        },
+        () => [ref.el?.querySelector(".o-autocomplete")]
```
Comment (FrancoisGe) [inline addons/web/static/src/core/record_selectors/tag_navigation_hook.js:27]: It doesnt work if the el arrive later ?

**Review by aab-odoo — APPROVED** [review]
robodoo rebase-ff r+

**Comment by aab-odoo** [comment]
robodoo r+

**Pull request closed without merging**